include/swift/Sema/ConstraintSystem.h

```diff
@@ -2177,6 +2177,11 @@ class ConstraintSystem {
   std::vector<std::pair<ConstraintLocator*, unsigned>>
       DisjunctionChoices;
 
+  /// A map from applied disjunction constraints to the corresponding
+  /// argument function type.
+  llvm::SmallMapVector<ConstraintLocator *, const FunctionType *, 4>
+      AppliedDisjunctions;
```

Comment:
What do you think about the alternative idea here of storing pointer to argument function in the disjunction locator?

Reply:
Oh, that would be much better. I'm going to merge this now (the release source compat failure looks unrelated), and see if I can make that change as a follow on cleanup. My only thought is we might not have the argument function at the time when the disjunction constraint is created